contracts/DCAFeeManager/DCAFeeManager.sol

```diff
@@ -39,40 +124,75 @@ contract DCAFeeManager is Governable, IDCAFeeManager {
   }
 
   /// @inheritdoc IDCAFeeManager
-  function setTargetTokensDistribution(TargetTokenShare[] calldata _newDistribution) external onlyOwnerOrAllowed {
-    _setTargetTokensDistribution(_newDistribution);
+  function resetAllowance(IERC20 _token) external {
+    _token.approve(address(hub), 0); // We do this first because some tokens (like USDT) will fail if we  don't
+    _token.approve(address(hub), type(uint256).max);
   }
 
-  function _setTargetTokensDistribution(TargetTokenShare[] memory _newDistribution) internal {
-    uint256 _currentTargetTokens = _distribution.length;
-    uint256 _min = _currentTargetTokens < _newDistribution.length ? _currentTargetTokens : _newDistribution.length;
+  /// @inheritdoc IDCAFeeManager
+  function availableBalances(address[] calldata _tokens) external view returns (AvailableBalance[] memory _balances) {
+    _balances = new AvailableBalance[](_tokens.length);
+    for (uint256 i; i < _tokens.length; i++) {
+      _balances[i] = AvailableBalance({
+        token: _tokens[i],
+        platformBalance: hub.platformBalance(_tokens[i]),
+        feeManagerBalance: IERC20(_tokens[i]).balanceOf(address(this))
+      });
+    }
+  }
 
-    uint16 _assignedShares;
-    for (uint256 i; i < _min; i++) {
-      // Rewrite storage
-      _assignedShares += _newDistribution[i].shares;
-      _distribution[i] = _newDistribution[i];
+  /// @inheritdoc IDCAFeeManager
+  function positionBalances(uint256[] calldata _positionIds) external view returns (PositionBalance[] memory _balances) {
+    _balances = new PositionBalance[](_positionIds.length);
+    for (uint256 i; i < _positionIds.length; i++) {
+      IDCAHubPositionHandler.UserPosition memory _userPosition = hub.userPosition(_positionIds[i]);
+      _balances[i] = PositionBalance({
+        positionId: _positionIds[i],
+        from: _userPosition.from,
+        to: _userPosition.to,
+        swappedBalance: _userPosition.swapped
+      });
     }
+  }
```

Comment:
Why not keep an array of created positions and also return that on `positionBalances` ?

Reply:
Done in https://github.com/Mean-Finance/dca-v2-periphery/pull/88